build.sc

```diff
@@ -650,30 +636,43 @@ object dev extends MillModule{
     val isWin = scala.util.Properties.isWin
     val millPath = T.ctx.dest / (if (isWin) "mill.bat" else "mill")
     os.move(super.assembly().path, millPath)
-    if (isWin) windowsVmOptions("dev.launcher", millPath, forkArgs())
     PathRef(millPath)
   }
 
   def prependShellScript = T{
+    val (millArgs, otherArgs) = forkArgs().partition(arg => arg.startsWith("-DMILL") && !arg.startsWith("-DMILL_VERSION"))
```

Comment:
We pass the `MILL_SOMETHING` properties via file, so that Windows doesn't complain!  
Everything except `MILL_VERSION` because it's used in lots of places.